### ISSUE TITLE
saving file changes syntax color

### ISSUE THREAD

**Issue by gucio321** [vim/vim#8081]
Hi there,
**Describe the bug**
I'm writing code in golang. after the last update I noticed, that syntax color changes after saving file `:w`

**To Reproduce**
1. open a `.go` file using vim (e.g. `vim main.go`)
2. type word declared as `goBuildins` in go.vim syntax e.g. `append`
3. save file (`:w`)
4. the color of `append` will be changed

**Expected behavior**
the color should not be changed (the second version of color should be displayed)

**Screenshots**
before `:w`
![image](https://user-images.githubusercontent.com/73652197/113860295-28cfbb80-97a6-11eb-8210-542c76f0a3a1.png)
after:
![image](https://user-images.githubusercontent.com/73652197/113860341-34bb7d80-97a6-11eb-9fcf-8a2f7bc58ca2.png)

**Environment (please complete the following information):**
 - Vim version:
```
 VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Mar 22 2021 00:00:00)
Included patches: 1-2637
Modified by <bugzilla@redhat.com>
Compiled by <bugzilla@redhat.com>
Huge version without GUI.  Features included (+) or not (-):
+acl               -farsi             +mouse_sgr         +tag_binary
+arabic            +file_in_path      -mouse_sysmouse    -tag_old_static
+autocmd           +find_in_path      +mouse_urxvt       -tag_any_white
+autochdir         +float             +mouse_xterm       -tcl
-autoservername    +folding           +multi_byte        +termguicolors
-balloon_eval      -footer            +multi_lang        +terminal
+balloon_eval_term +fork()            -mzscheme          +terminfo
-browse            +gettext           +netbeans_intg     +termresponse
++builtin_terms    -hangul_input      +num64             +textobjects
+byte_offset       +iconv             +packages          +textprop
+channel           +insert_expand     +path_extra        +timers
+cindent           +ipv6              +perl/dyn          +title
-clientserver      +job               +persistent_undo   -toolbar
-clipboard         +jumplist          +popupwin          +user_commands
+cmdline_compl     +keymap            +postscript        +vartabs
+cmdline_hist      +lambda            +printer           +vertsplit
+cmdline_info      +langmap           +profile           +virtualedit
+comments          +libcall           -python            +visual
+conceal           +linebreak         +python3/dyn       +visualextra
+cryptv            +lispindent        +quickfix          +viminfo
+cscope            +listcmds          +reltime           +vreplace
+cursorbind        +localmap          +rightleft         +wildignore
+cursorshape       +lua/dyn           +ruby/dyn          +wildmenu
+dialog_con        +menu              +scrollbind        +windows
+diff              +mksession         +signs             +writebackup
+digraphs          +modify_fname      +smartindent       -X11
-dnd               +mouse             -sound             -xfontset
-ebcdic            -mouseshape        +spell             -xim
+emacs_tags        +mouse_dec         +startuptime       -xpm
+eval              +mouse_gpm         +statusline        -xsmp
+ex_extra          -mouse_jsbterm     -sun_workshop      -xterm_clipboard
+extra_search      +mouse_netterm     +syntax            -xterm_save
   system vimrc file: "/etc/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/etc"
 f-b for $VIMRUNTIME: "/usr/share/vim/vim82"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 
Linking: gcc -L. -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector-strong -rdynamic -Wl,-export-dynamic -Wl,--enable-new-dtags -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -L/usr/local/lib -Wl,--as-needed -o vim -lm -lselinux -lncurses -lacl -lattr -lgpm -ldl -Wl,--enable-new-dtags -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector-strong -L/usr/local/lib -L/usr/lib64/perl5/CORE -lperl -lpthread -lresolv -ldl -lm -lcrypt -lutil -lc 
```

 - OS: Fedora release 33 (Thirty Three)
 - Terminal: # GNOME Terminal 3.38.1 using VTE 0.62.3 +BIDI +GNUTLS +ICU +SYSTEMD

**Additional Context**
Let me know, if that's a wrong place to report this (e.g. that's redhat's fault)

**Comment by chrisbra** [comment]
does it happen with `vim --clean`?

**Comment by gucio321** [comment]
> does it happen with vim --clean?

no

Just tried to debug my .vimrc a bit, and I can see, that this bug is caused by
```
color delek
```
so that's fault of a single color-set

**Comment by lacygoill** [comment]
> this bug is caused by
>
>     color delek

I still can't reproduce:

    vim -Nu NONE -S <(cat <<'EOF'
        filetype plugin indent on
        syntax on
        colorscheme delek
        let lines =<< END
        if t == d2ds1.WallLayerGroup {
                l = append(l,
    END
        call writefile(lines, '/tmp/go.go')
        silent edit /tmp/go.go
        call search('append')
        function LogAttributeUnderCursor()
            let msg = 'attribute of cell under cursor BEFORE writing: ' .. screenattr(winline(), wincol())
            call writefile([msg], '/tmp/log')
            silent write
            let msg = 'attribute of cell under cursor AFTER writing: ' .. screenattr(winline(), wincol())
            call writefile([msg], '/tmp/log', 'a')
            qa!
        endfunction
        call timer_start(0, {-> LogAttributeUnderCursor()})
    EOF
    ) && cat /tmp/log

Gives:

    attribute of cell under cursor BEFORE writing: 311
    attribute of cell under cursor AFTER writing: 311

**Comment by gucio321** [comment]
yah, your code works for me too.

could you try `vim --cmd "color delek" main.go`

**Comment by lacygoill** [comment]
Same result, the color of the `append` function is preserved after writing the buffer:

![gif](https://user-images.githubusercontent.com/8505073/113888331-896cf180-97c2-11eb-95de-c2966923cae7.gif)

**Comment by gucio321** [comment]
oh, I found the bug. I was using vim-go plugin,
first, I installed it in the simplest way (copied to .vim)
and then started to use vundle, but I forgot to remove my old vim-go from .vim.
I disabled vundle's version of vim-go in my .vimrc for testing time, but just noticed an old version.
and because vim-go changes (improves) coloring, my bug was caused by it.
So: wrong repo :grinning:
Thanx for help @lacygoill @chrisbra
